c-c++.md

```diff
@@ -31,6 +31,7 @@ Starred version marks the default system compiler.
 
 Distribution    | GCC                  | Clang/LLVM
 ----------------|----------------------|-------------
+Amazon Linux 2023  | 11*               | 15*
```

Comment:
The asterisks are weird here, but it's already an established norm for other rows